homepage/homepage/app/docs/[framework]/[...slug]/schemas/accounts-and-migrations.mdx

```diff
@@ -0,0 +1,213 @@
+
+  return <div>
+    <h1>Dashboard</h1>
+    {me ? <div>
+      <p>Logged in as {me.profile.name}</p>
+      <h2>My chats</h2>
+      {me.root.myChats.map((chat) => <ChatPreview key={chat.id} chat={chat} />)}
+      <h2>My contacts</h2>
+      {me.root.myContacts.map((contact) => <ContactPreview key={contact.id} contact={contact} />)}
+    </div> : "Loading..."}
+  </div>
+}
+
+```
+</CodeGroup>
+</ContentByFramework>
+
+
+## Populating and evolving `root` and `profile` schemas with migrations
+
+As you develop your app, you'll likely want to
+
+- initialise data in a user's `root` and `profile`
+- add more data to your `root` and `profile` schemas
+
+You can achieve both by overriding the static `migrate()` method on your `Account` schema class.
+
+### When migrations run
+
+Migrations are run after account creation and every time a user logs in.
+Jazz waits for the migration to finish before passing the account to your app's context.
+
+### Initialising user data after account creation
+
+<CodeGroup>
+{/* prettier-ignore */}
+```ts
+export class MyAppAccount extends Account {
+  root = co.ref(MyAppRoot);
+
+  async migrate() {
+    const me = this;
+
+// we specifically need to check for undefined, because the root might simply be not loaded (`null`) yet
+    if (me.root === undefined) {
+      me.root = MyAppRoot.create({
+        // Using a group to set the owner is always a good idea.
+        // This way if in the future we want to share this coValue we can do so easily.
+        myChats: ListOfChats.create([], Group.create(me)),
+        myContacts: ListOfAccounts.create([], Group.create(me))
+      }, Group.create(me));
+    }
+  }
+}
+```
+</CodeGroup>
+
+### Adding/changing fields to `root` and `profile`
+
+To add new fields to your `root` or `profile` schemas, amend their corresponding schema classes with new fields,
+and then implement a migration that will populate the new fields for existing users (by using initial data, or by using existing data from old fields).
+
+To do deeply nested migrations, you might need to use the asynchronous `ensureLoaded()` method before determining whether the field already exists, or is simply not loaded yet.
+
+Now let's say we want to add a `myBookmarks` field to the `root` schema:
+
+<CodeGroup>
+{/* prettier-ignore */}
+```ts
+export class MyAppAccount extends Account {
+    root = co.ref(MyAppRoot);// old
+
+    async migrate() { // old
+      const me = this; // old
+
+      if (me.root === undefined) {  // old
+        me.root = MyAppRoot.create({  // old
+          myChats: ListOfChats.create([], Group.create(me)),  // old
+          myContacts: ListOfAccounts.create([], Group.create(me)) // old
+        }, Group.create(me)); // old
+      } // old
+
+      // We need to load the root field to check for the myContacts field
+      const { root } = await me.ensureLoaded({
+        root: {},
+      });
+
+      if (!root) throw new Error("Root missing!"); // this should never happen
```

Comment:
We need a `ensureLoaded` that throws if the value isn't available.

This way the user code gets much easier.